src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -955,6 +955,18 @@ void primitive_inst::realloc_if_needed(bool prev_execution_skipped) {
                                           is_output_buffer(this, true),
                                           output_memory_ptr(i).get(),
                                           true);
+            // memory_pool does not reset padded memory to be reused.
```

Comment:
Can this be combined with the logic from `can_reuse_buffer` branch?

Reply:
It doesn't seem easy. `can_reuse_buffer` still works properly, but it causes problems when the input shape keeps changing randomly, rather than when the buffer size keeps increasing.